content/hacking-atom/sections/hacking-on-atom-core.md

```diff
@@ -0,0 +1,307 @@
+
+* `'node-gyp' is not recognized as an internal or external command, operable program or batch file.`
+  * Try running `npm install -g node-gyp`, and run `script\build` again.
+
+* Other `node-gyp` errors on first build attempt, even though the right Node.js and Python versions are installed.
+  * Do try the build command one more time as experience shows it often works on second try in many cases.
+
+###### Windows build error reports
+
+* If all else fails, use [this search](https://github.com/atom/atom/search?q=label%3Abuild-error+label%3Awindows&type=Issues) to get a list of reports about build errors on Windows, and see if yours has already been reported.
+* If it hasn't, please open a new issue with your Windows version, architecture (x86 or x64), and a text dump of your build output, including the Node.js and Python versions.
+
+{{/windows}}
+
+{{#linux}}
+
+###### TypeError: Unable to watch path
+
+If you get following error with a big traceback right after Atom starts:
+
+```
+TypeError: Unable to watch path
+```
+
+you have to increase number of watched files by inotify.  For testing if this is the reason for this error you can execute:
+
+``` command-line
+$ sudo sysctl fs.inotify.max_user_watches=32768
+```
+
+then restart Atom.  If Atom now works fine, you can make this setting permanent:
+
+``` command-line
+$ echo 32768 | sudo tee -a /proc/sys/fs/inotify/max_user_watches
+```
+
+See also [#2082](https://github.com/atom/atom/issues/2082).
+
+###### /usr/bin/env: node: No such file or directory
+
+If you get this notice when attempting to run any script, you either do not have
+Node.js installed, or node isn't identified as Node.js on your machine. If it's
+the latter, this might be caused by installing Node.js via the distro package
+manager and not nvm, so entering `sudo ln -s /usr/bin/nodejs /usr/bin/node` into
+your terminal may fix the issue. On some variants (mostly Debian based distros)
+you can use `update-alternatives` too:
+
+``` command-line
+$ sudo update-alternatives --install /usr/bin/node node /usr/bin/nodejs 1 --slave /usr/bin/js js /usr/bin/nodejs
+```
+
+###### AttributeError: 'module' object has no attribute 'script_main'
+
+If you get following error with a big traceback while building Atom:
+
+```
+sys.exit(gyp.script_main()) AttributeError: 'module' object has no attribute 'script_main' gyp ERR!
+```
+
+you need to uninstall the system version of gyp.
+
+On Fedora you would do the following:
+
+``` command-line
+$ sudo yum remove gyp
+```
+
+###### Linux build error reports
+
+Use [this search](https://github.com/atom/atom/search?q=label%3Abuild-error+label%3Alinux&type=Issues)
+to get a list of reports about build errors on Linux.
+
+{{/linux}}
+
+#### Running in Development Mode
```

Comment:
✨  to this whole section. Should you link to it from the blurb at the top since there's a ton of content about building in between?

Reply:
Do you have to have built at least once before running in Development Mode will work the way it is documented here? Or will it work, just the first startup will be slow because it will have to compile everything?

Reply:
You *need* to have the dependencies installed via `script/bootstrap` but you don't have to have it built. No matter what the first run is going to take a while to warm the cache, since I'm pretty sure we don't share caching between the build and dev mode.

Reply:
I moved the build section to the end ... this way just cloning and running in development mode is nice and streamlined and they can read the full build instructions later.